lockfile/src/parsers/go_sum.rs

```diff
@@ -0,0 +1,51 @@
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+
+use super::*;
+
+pub fn parse(input: &str) -> Result<&str, Vec<PackageDescriptor>> {
+    let pkgs = input.lines().filter_map(package).collect::<Vec<_>>();
+    Ok((input, pkgs))
+}
+
+fn package(input: &str) -> Option<PackageDescriptor> {
+    let (input, name) = package_name(input).ok()?;
+    let (input, version) = package_version(input).ok()?;
+    let (_, _hash) = package_hash(input).ok()?;
+
+    // If the package version ends in "/go.mod" then this entry
+    // just records the hash for a go.mod file. The package this
+    // go.mod file belongs to will also be listed, and that's what
+    // we're interested in, so we simply discard this entry.
+    if version.ends_with(r"/go.mod") {
```

Comment:
Could use `.then(|| ...)` if you want to be fancy.

Reply:
love to be fancy